sklearn/model_selection/_search.py

```diff
@@ -242,16 +243,36 @@ class ParameterSampler:
     True
     """
     def __init__(self, param_distributions, n_iter, random_state=None):
-        self.param_distributions = param_distributions
+        if not isinstance(param_distributions, (Mapping, Iterable)):
+            raise TypeError('Parameter distribution is not a dict or '
+                            'a list ({!r})'.format(param_distributions))
+
+        if isinstance(param_distributions, Mapping):
+            # wrap dictionary in a singleton list to support either dict
+            # or list of dicts
+            param_distributions = [param_distributions]
+
+        for dist in param_distributions:
+            if not isinstance(dist, dict):
+                raise TypeError('Parameter distribution is not a '
+                                'dict ({!r})'.format(dist))
+            for key in dist:
+                if (not isinstance(dist[key], Iterable)
+                        and not hasattr(dist[key], 'rvs')):
+                    raise TypeError('Parameter value is not iterable '
```

Comment:
... must be an iterable or a distribution?

Reply:
this is copy & pasted from ParameterGrid. Not sure if your version is any clearer and I think being semi-consistent between the two is good.